File: backend/app/schemas/backtest.py
from pydantic import BaseModel, Field


class BacktestCreateRequest(BaseModel):
    strategy: str = Field(min_length=1, description="策略名称")
    timeframe: str = Field(min_length=1, description="时间周期")
    timerange: str = Field(min_length=1, description="回测时间范围")
    pairs: list[str] = Field(min_length=1, description="交易对列表")
    stake_amount: float = Field(gt=0, description="单次投入资金")
    fee: float = Field(ge=0, description="手续费")
    config_preset: str = Field(min_length=1, description="配置预设")


class BacktestJobSummary(BaseModel):
    id: str
    status: str
    strategy: str
    timeframe: str
    timerange: str
    pairs: list[str]
    stake_amount: float
    fee: float
    config_preset: str

File: frontend/src/types/backtest.ts
export type BacktestJob = {
  id: string;
  status: string;
  strategy: string;
  timeframe: string;
  timerange: string;
  pairs: string[];
  stake_amount: number;
  fee: number;
  config_preset: string;
};

export type KlineCandle = {
  time: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
};

export type HistoricalDataRequest = {
  exchange: string;
  pairs: string[];
  timeframes: string[];
  timerange: string;
};

export type HistoricalDataResponse = {
  status: string;
  command: string[];
  stdout: string;
  stderr: string;
  data_path: string;
  used_fallback: boolean;
};
